src/components/entity/resolve.vue

```diff
@@ -0,0 +1,144 @@
+
+This file is part of ODK Central. It is subject to the license terms in
+the LICENSE file found in the top-level directory of this distribution and at
+https://www.apache.org/licenses/LICENSE-2.0. No part of ODK Central,
+including this file, may be copied, modified, propagated, or distributed
+except according to the terms contained in the LICENSE file.
+-->
+<template>
+  <modal id="entity-resolve" :state="state" :hideable="!awaitingResponse" large
+    backdrop @shown="afterShown" @hide="$emit('hide')">
+    <template #title>{{ $t('title', props.entity) }}</template>
+    <template #body>
+      <div v-if="!success">
+        <p>{{ $t('instructions[0]', props.entity) }}</p>
+        <p>{{ $t('instructions[1]', { markAsResolved: $t('action.markAsResolved') }) }}</p>
+
+        <!-- placeholder for summary table -->
+
+        <router-link class="btn btn-default more-details" :to="entityPath(dataset.projectId, dataset.name, props.entity?.__id)"
+          target="_blank">
+          <span class="icon-external-link-square"></span>{{ $t('action.seeMoreDetails') }}
+        </router-link>
+        <button type="button" class="btn btn-default">
+          <span class="icon-pencil"></span>{{ $t('action.editEntity') }}
+        </button>
+        <button type="button" class="btn btn-default mark-as-resolved" :aria-disabled="awaitingResponse" @click="markAsResolve">
+          <span class="icon-check"></span>{{ $t('action.markAsResolved') }} <spinner :state="awaitingResponse"/>
+        </button>
+</div>
+      <div v-else class="success-msg">
+        <span class="icon-check-circle success"></span> {{ $t('successMessage') }}
+      </div>
+
+      <div class="modal-actions">
+        <button type="button" class="btn btn-primary" @click="$emit('hide')">
+          {{ $t('action.close') }}
+        </button>
+      </div>
+    </template>
+  </modal>
+</template>
+
+<script setup>
+import { ref } from 'vue';
+import { useI18n } from 'vue-i18n';
+
+import useRoutes from '../../composables/routes';
+import useRequest from '../../composables/request';
+
+import Modal from '../modal.vue';
+import Spinner from '../spinner.vue';
+
+import { apiPaths } from '../../util/request';
+import { noop } from '../../util/util';
+import { useRequestData } from '../../request-data';
+
+defineOptions({
+  name: 'EntityResolve'
+});
+
+const props = defineProps({
+  state: Boolean,
+  entity: Object
+});
+
+const emit = defineEmits(['hide', 'success']);
+
+const { t } = useI18n();
+const { request, awaitingResponse } = useRequest();
+const { dataset } = useRequestData();
+const { entityPath } = useRoutes();
+
+const success = ref(false);
+
+const afterShown = () => {
+  success.value = false;
+};
+
+const markAsResolve = () => {
+  const { entity } = props;
+  const url = apiPaths.entity(dataset.projectId, dataset.name, entity.__id, { resolve: true });
+
+  request.patch(
+    url,
+    null,
+    {
+      headers: { 'If-Match': `"${entity.__system.version}"` },
```

Comment:
I need to change backend to handle this

Reply:
Good thought, I think that makes a lot of sense to require the user to specify the version number when resolving.